src/transports/smart_protocol.c

```diff
@@ -622,7 +622,9 @@ int git_smart__download_pack(
 			}
 		}
 
-		git__free(pkt);
+		if (pkt) {
+			git_pkt_free(pkt);
```

Comment:
Huh, one more time. I know it's not your fault, but instead the existing code is to blame. But I'd highly appreciate if you did a preliminary commit which fixes `git_pkt_free` to return early in case a `NULL` pointer is being passed to it.